### PR TITLE
refactor (jkube-kit-resource-service): Migrate test from JUnit4 to JUnit5 (#1567)

### LINKED ISSUE #1567: (jkube-kit-resource-service): Replace JUnit4 with JUnit5
### Component

jkube-kit-resource-service

### Task description

Child of #1546

- Remove JUnit4 dependencies
- Add JUnit5 dependencies
- Refactor tests

### PR DESCRIPTION
**Pull request by anurag-rajawat**
## Description

Fixes #1567 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/DefaultResourceServiceTest.java [hunk 1]
```diff
@@ -60,22 +55,22 @@ public class DefaultResourceServiceTest {
   private ResourceServiceConfig resourceServiceConfig;
   private DefaultResourceService defaultResourceService;
 
-  @Before
-  public void init() throws IOException {
-    targetDir = temporaryFolder.newFolder("target");
+  @BeforeEach
+  void init(@TempDir File temporaryFolder) {
+    targetDir = new File(temporaryFolder, "target");
```
Comment (manusa) [inline jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/DefaultResourceServiceTest.java:60]: We should create/link an actual directory

File: jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/DefaultResourceServiceTest.java [hunk 2]
```diff
@@ -60,22 +55,22 @@ public class DefaultResourceServiceTest {
...
     resourceServiceConfig = ResourceServiceConfig.builder()
         .interpolateTemplateParameters(true)
         .targetDir(targetDir)
         .project(project)
         .resourceFileType(ResourceFileType.yaml)
-        .resourceDirs(Collections.singletonList(temporaryFolder.newFolder("resources")))
+        .resourceDirs(Collections.singletonList(new File(temporaryFolder, "resources")))
```
Comment (manusa) [inline jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/DefaultResourceServiceTest.java:66]: We should use an actual directory, not just a reference to a nonexistent one

File: jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java [hunk 3]
```diff
@@ -69,9 +65,9 @@ public void writeResource() throws IOException {
   }
 
   @Test
-  public void writeResourceThrowsException() throws IOException {
+  void writeResourceThrowsException() throws IOException {
     // Given
-    final File resource = temporaryFolder.newFolder("resource-base");
+    final File resource = new File(temporaryFolder, "resource-base");
```
Comment (manusa) [inline jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java:70]: You should use an existent directory

**Review by manusa — CHANGES_REQUESTED** [review]
Looks good except for the minor comments

File: jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/DefaultResourceServiceTest.java (hunk 1 above)
Comment (anurag-rajawat) [reply jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/DefaultResourceServiceTest.java:60]: Got it

File: jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/DefaultResourceServiceTest.java (hunk 2 above)
Comment (anurag-rajawat) [reply jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/DefaultResourceServiceTest.java]: Got it

File: jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java (hunk 3 above)
Comment (anurag-rajawat) [reply jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java]: Got it

**Review by manusa — APPROVED** [review]

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**